### DISCUSSION TITLE
Writing a function to find electrophiles

### DISCUSSION THREAD

**Discussion by rishigurnani** [rdkit/rdkit#6468] [Q&A]
I'm trying to write a function to identify electrophiles using the definition from [this link](https://chem.libretexts.org/Bookshelves/Organic_Chemistry/Organic_Chemistry_I_(Cortes)/12%3A_Introduction_to_Lewis_Acid-Base_Chemistry/12.03%3A_Recognizing_Electrophiles#:~:text=Electrophilic%20centers%20are%20areas%20of,by%20identifying%20a%20polar%20bond.):
> Electrophilic centers are areas of low electron density. Most often they are atoms which (a) contain an incomplete octet, and/or (b) carry a full or a partial positive charge. A partial positive charge can be revealed by writing resonance structures, or by identifying a polar bond.

As an intermediate step, I'm just trying to write a function to classify a molecule as belonging to (b) or not. I tried the following functions:
```python
def getResonanceStructures(mol, flags = 0):
    suppl = Chem.ResonanceMolSupplier(mol, flags)
    resMols = [mol for mol in suppl]
    return resMols

def is_electrophile(mol):
  my_match = f"[*+1]"
  res_mols = getResonanceStructures(mol_without_atom_index(mol),
                  flags = Chem.ResonanceFlags.ALLOW_INCOMPLETE_OCTETS
                  | Chem.ResonanceFlags.UNCONSTRAINED_CATIONS
                  | Chem.ResonanceFlags.UNCONSTRAINED_ANIONS)
  match_mol = Chem.MolFromSmarts(my_match)
  return any([x.HasSubstructMatch(match_mol) for x in res_mols])
```
But it fails for `CC#N`. This was unexpected because [this link](https://chem.libretexts.org/Courses/Athabasca_University/Chemistry_360%3A_Organic_Chemistry_II/Chapter_20%3A_Carboxylic_Acids_and_Nitriles/20.07_Chemistry_of_Nitriles) says that the molecule is an electrophile because it has a resonance structure with a positive charge.
![image](https://github.com/rdkit/rdkit/assets/13811245/0178884a-bf3a-491c-bc9e-6959155a3b85)

Any help would be greatly appreciated. Thanks in advance!

**Comment by greglandrum** [comment]
@rishigurnani please provide a code sample which we can actually run. We don't know what `mol_without_atom_index()` does

**Reply by rishigurnani** [reply]
Sorry about that! I didn't see that `mol_without_atom_index` was hidden in there. Here is a revised version which I ran locally.

```python
from rdkit import Chem

def getResonanceStructures(mol, flags = 0):
    suppl = Chem.ResonanceMolSupplier(mol, flags)
    resMols = [mol for mol in suppl]
    return resMols

def is_electrophile(mol):
  my_match = f"[*+1]"
  res_mols = getResonanceStructures(mol,
                  flags = Chem.ResonanceFlags.ALLOW_INCOMPLETE_OCTETS
                  | Chem.ResonanceFlags.UNCONSTRAINED_CATIONS
                  | Chem.ResonanceFlags.UNCONSTRAINED_ANIONS)
  match_mol = Chem.MolFromSmarts(my_match)
  return any([x.HasSubstructMatch(match_mol) for x in res_mols])

sm = "CC#N"
mol = Chem.MolFromSmiles(sm)
result = is_electrophile(mol)
print(result)
```
`False` is printed but I would have expected `True` based on the discussion in my initial post.
